src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -176,7 +183,11 @@ public void setSkipIfLastAndSharedWithCase(boolean newValue) {
     @Override
     public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
-        if (skipIfLastAndSharedWithCase) {
+
+        // Switch rules are not subject to fall through.
+        final boolean isSwitchRule = defaultGroupAST.getType() == TokenTypes.SWITCH_RULE;
+
+        if (skipIfLastAndSharedWithCase && !isSwitchRule) {
```

Comment:
Not related to this PR - the name "skipIfLastAndSharedWithCase" seems a little confusing to me... or, the logic seems to be "skip if (NOT last) and (shared with case)", but I am probably missing something?

Reply:
This is a public [property](https://checkstyle.sourceforge.io/config_coding.html#DefaultComesLast_Properties), so fixing the name will be a breaking change.

Reply:
I understand, I didn't mean to request it should be changed, just an observation.